cli/src/cli_util.rs

```diff
@@ -3062,6 +3063,104 @@ impl fmt::Display for RemoteBookmarkNamePattern {
     }
 }
 
+/// Computes the location (new parents and new children) to place commits.
+///
+/// The `destination` argument is mutually exclusive to the `insert_after` and
+/// `insert_before` arguments.
+pub fn compute_commit_location(
+    ui: &mut Ui,
+    workspace_command: &mut WorkspaceCommandHelper,
+    destination: &Option<Vec<RevisionArg>>,
+    insert_after: &Option<Vec<RevisionArg>>,
+    insert_before: &Option<Vec<RevisionArg>>,
+    commit_description: &str,
+) -> Result<(Vec<Commit>, Vec<Commit>), CommandError> {
```

Comment:
Is it better to return lists of commit IDs? Many callers don't seem to need the commit objects, so it's better to not load the objects here if not necessary. (Since the `Store` caches commits, it's less of a problem to unnecessarily load a commit twice, if we were to load some commit here and then again in the caller.)

Reply:
I think that's best. I don't think the extra reading matters much when the commit is cached. Reading extra commits is probably a bigger problem, especially with slow backend. It may make a difference on the server at Google. But let me know if you think it gets to messy.

Reply:
It seems like the main reason I returned the commits is that `workspace_command.resolve_some_revsets_default_single` also returned commit objects. Maybe we should create a version of the function that only returns commit IDs instead?